### PR TITLE
performance_schema: code refactoring and remove old implementation

### PR DESCRIPTION
**Pull request by nieyy**
Code refactoring:
1. remove the implementation from `plan/plans`.
2. add `perfschema/const.go` and `perfschema/common/go`

PTAL, @shenli @coocood

### CODE REVIEW

File: perfschema/common.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// castValues casts values based on columns type.
+func castValues(rec []interface{}, cols []*model.ColumnInfo) (err error) {
```
Comment (qiuyesuifeng) [inline perfschema/common.go]: Why not using column.CastValues instead?

Comment (nieyy) [inline perfschema/common.go]: If I did so, actually I have to add more code, you can take a look in later PRs.

File: perfschema/common.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+// checkNotNull checks if nil value set to a column with NotNull flag is set.
+func checkNotNull(c *model.ColumnInfo, data interface{}) error {
+	if mysql.HasNotNullFlag(c.Flag) && data == nil {
+		return errors.Errorf("Column %s can't be null.", c.Name)
+	}
+	return nil
+}
+
+// checkNotNulls checks if row has nil value set to a set of columns with NotNull flag set.
+func checkNotNulls(cols []*model.ColumnInfo, row []interface{}) error {
```
Comment (qiuyesuifeng) [inline perfschema/common.go]: How about using column.CheckNotNull?

Comment (nieyy) [inline perfschema/common.go]: I am not sure if this is a problem, but I suggest we can revise it in later PRs if you really dislike this decision, is that ok?

File: perfschema/common.go (hunk 1 above)
Comment (qiuyesuifeng) [inline perfschema/common.go]: Gotcha, we should later unify column.Col and model.ColumnInfo, then it will be much clearer.

**Comment by coocood** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**